src/mlpack/methods/ne/cne.hpp

```diff
@@ -0,0 +1,186 @@
+ /**
+ * @file cne.hpp
+ * @author Bang Liu
+ *
+ * Definition of CNE class.
+ */
+#ifndef MLPACK_METHODS_NE_CNE_HPP
+#define MLPACK_METHODS_NE_CNE_HPP
+
+#include <cstddef>
+#include <cstdio>
+
+#include <mlpack/core.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "genome.hpp"
+#include "species.hpp"
+#include "tasks.hpp"
+#include "parameters.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class implements Conventional Neuro-evolution (CNE): weight
+ * evolution on topologically fixed neural networks.
+ */
+template<typename TaskType>
+class CNE {
+ public:
+  // Parametric constructor.
+  CNE(TaskType task, Genome& seedGenome, Parameters& params) {
```

Comment:
It would be great if you could provide a Constructor that allows a user to directly set the parameters, instead of passing a parameter object.